### PR TITLE
Add skeleton import functionality to mesh_loader

### PR DESCRIPTION
**Pull request by kar-re**
### Description

This adds the functionality of importing an animated mesh to RViz using the same resource_retriever as in mesh_loader. 

Due to the way Ogre3D loads skeleton meshes a developer can automatically import an animated mesh using the same loadMeshFromResource method already present, but with the added functionality of assigning animation states.
The method is mostly the same as loadMeshFromResource but with changes to load the .skeleton file as well into the default resource group. 

Happy to accept any constructive critiques or suggestions on how to make it better!

### Checklist

- [ ] If you are addressing rendering issues, please provide:
  - [ ] Images of both, broken and fixed renderings.
  - [ ] Source code to reproduce the issue, e.g. a `YAML` or `rosbag` file with a `MarkerArray` msg.
- [ ] If you are changing GUI, please include screenshots showing how things looked *before* and *after*.
- [ ] Choose the proper target branch: latest release branch, for non-ABI-breaking changes, *future* release branch otherwise.
      Due to the lack of active maintainers, we cannot provide support for older release branches anymore.
- [Will do if accepted! ] Did you change how RViz works? Added new functionality? Do not forget to update the tutorials and/or documentation on the [ROS wiki](http://wiki.ros.org/rviz)
- [x] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-visualization/rviz/pulls) to support the maintainers of RViz. Refer to the [RViz Wiki](https://github.com/ros-visualization/rviz/wiki/Maintainer-Guide#reviewing-pull-requests) for reviewing guidelines.

### CODE REVIEW

File: src/rviz/mesh_loader.cpp [hunk 1]
```diff
@@ -699,6 +704,7 @@ Ogre::MeshPtr loadMeshFromResource(const std::string& resource_path)
       {
         return Ogre::MeshPtr();
       }
+      Ogre::SkeletonPtr skeleton = loadSkeletonFromResource(resource_path);
```
Comment (rhaschke) [inline src/rviz/mesh_loader.cpp:707]: Where / how is this `skeleton` used?

**Review by rhaschke — COMMENTED** [review]
I don't yet see the use of this PR. Could you explain more, please?
According to the [OGRE doc](https://www.ogre3d.org/docs/api/1.9/class_ogre_1_1_skeleton.html#details), skeleton definitions are loaded on demand, especially when referenced by a Mesh. So, do we really need to explicitly load them?

How do you intend animating a mesh based on the skeleton? The `SkeletonPtr` isn't used yet.

File: src/rviz/mesh_loader.cpp (hunk 1 above)
Comment (rhaschke) [reply src/rviz/mesh_loader.cpp:707]:
```suggestion
      loadSkeletonFromResource(resource_path);  // load skeleton to the resource manager
```

File: src/rviz/mesh_loader.cpp [hunk 2]
```diff
@@ -699,6 +704,7 @@ Ogre::MeshPtr loadMeshFromResource(const std::string& resource_path)
       {
         return Ogre::MeshPtr();
       }
+      loadSkeletonFromResource(resource_path);  // load skeleton to the resource manager
```
Comment (rhaschke) [inline src/rviz/mesh_loader.cpp:707]:
```suggestion
      loadSkeletonFromResource(resource_path); // load skeleton to the resource manager
```

**Comment by kar-re** [comment]
Hi @rhaschke, just wanted to say thanks for implementing the code review change, and sorry for the late response. Please let me know if there's anything else I can do! :)

**Pull request merged**